src/Raven.Client/Documents/BulkInsert/BulkInsertOperation.cs

```diff
@@ -263,6 +269,51 @@ public BulkInsertOperation(string database, IDocumentStore store, BulkInsertOpti
 
             _generateEntityIdOnTheClient = new GenerateEntityIdOnTheClient(_requestExecutor.Conventions,
                 entity => AsyncHelpers.RunSync(() => _requestExecutor.Conventions.GenerateDocumentIdAsync(database, entity)));
+
+            _streamLock = new AsyncReaderWriterLock();
```

Comment:
Wouldn't it be simpler to use `SemaphoreSlim(1,1)` ? I'm not sure what is the performance of 'AsyncReaderWriterLock', but probably it is worse than `SemaphoreSlim`  ?

Reply:
done